core/server/src/main/java/alluxio/worker/netty/NettyDataServer.java

```diff
@@ -66,11 +66,11 @@ public void close() throws IOException {
     int quietPeriodSecs =
         Configuration.getInt(PropertyKey.WORKER_NETWORK_NETTY_SHUTDOWN_QUIET_PERIOD);
     int timeoutSecs = Configuration.getInt(PropertyKey.WORKER_NETWORK_NETTY_SHUTDOWN_TIMEOUT);
-    // TODO(binfan): investigate when timeoutSecs is zero (e.g., set in integration tests), does
-    // this still complete successfully.
     mChannelFuture.channel().close().awaitUninterruptibly(timeoutSecs, TimeUnit.SECONDS);
-    mBootstrap.group().shutdownGracefully(quietPeriodSecs, timeoutSecs, TimeUnit.SECONDS);
-    mBootstrap.childGroup().shutdownGracefully(quietPeriodSecs, timeoutSecs, TimeUnit.SECONDS);
+    mBootstrap.group().shutdownGracefully(quietPeriodSecs, timeoutSecs, TimeUnit.SECONDS)
+        .awaitUninterruptibly(timeoutSecs, TimeUnit.SECONDS);
+    mBootstrap.childGroup().shutdownGracefully(quietPeriodSecs, timeoutSecs, TimeUnit.SECONDS)
```

Comment:
added return value checking ... I agree with you that waiting up to three times the time out value is a bit odd, but I was not able to find API that would allow us to group the wait ... do you have any suggestion?